tajo-catalog/tajo-catalog-common/src/main/java/org/apache/tajo/catalog/CatalogService.java

```diff
@@ -162,7 +162,17 @@ PartitionDescProto getPartition(String databaseName, String tableName, String pa
       throws UndefinedPartitionException, UndefinedPartitionMethodException, UndefinedDatabaseException,
       UndefinedTableException;
 
-  List<PartitionDescProto> getPartitions(String databaseName, String tableName);
+  List<PartitionDescProto> getAllPartitions(String databaseName, String tableName) throws UndefinedDatabaseException,
+    UndefinedTableException, UndefinedPartitionMethodException, UndefinedPartitionException, UndefinedOperatorException;
+
+  boolean existPartitions(String databaseName, String tableName) throws UndefinedDatabaseException,
+  UndefinedTableException, UndefinedPartitionMethodException;
+
+  List<PartitionDescProto> getPartitionsByAlgebra(PartitionsByAlgebraProto request) throws
+    UndefinedDatabaseException, UndefinedTableException, UndefinedPartitionMethodException;
+
+  List<PartitionDescProto> getPartitionsByDirectSql(PartitionsByDirectSqlProto request) throws
```

Comment:
Its name should be changed into {{PartitionByFilter}} or {{PartitionByPredicate}}. As far as I know, the suffix ``DirectSQL`` is inspired by Hive API. But, the Hive API does predefined filter expression instead of direct sql predicate. Direct SQL is always danger, and this is not our way.